x-pack/plugins/security_solution/server/endpoint/services/agent/clients/sentinel_one/sentinel_one_agent_status_client.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AggregationsAggregate, SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+import { HostStatus } from '../../../../../../common/endpoint/types';
+import type { AgentStatuses, RawSentinelOneInfo } from '../lib/types';
+import type { ResponseActionAgentType } from '../../../../../../common/endpoint/service/response_actions/constants';
+import { AgentStatusClient } from '../lib/base_agent_status_client';
+import { AgentStatusClientError } from '../errors';
+
+const SENTINEL_ONE_AGENT_INDEX = `logs-sentinel_one.agent-default`;
+
+enum SENTINEL_ONE_NETWORK_STATUS {
+  CONNECTING = 'connecting',
+  CONNECTED = 'connected',
+  DISCONNECTING = 'disconnecting',
+  DISCONNECTED = 'disconnected',
+}
+
+export class SentinelOneAgentStatusClient extends AgentStatusClient {
+  protected readonly agentType: ResponseActionAgentType = 'sentinel_one';
+
+  async getAgentStatuses(agentIds: string[]): Promise<AgentStatuses> {
+    const dateField = 'sentinel_one.agent.last_active_date';
+
+    const query = {
+      bool: {
+        must: [
+          {
+            bool: {
+              filter: [
+                {
+                  terms: {
+                    'sentinel_one.agent.uuid': agentIds,
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      },
+    };
+
+    try {
+      const searchResult: SearchResponse<
+        RawSentinelOneInfo,
+        Record<string, AggregationsAggregate>
+      > = await this.options.esClient.search(
+        {
+          index: SENTINEL_ONE_AGENT_INDEX,
+          size: 10000,
+          query,
+          aggs: {
```

Comment:
Changed in 7431c9caae6d4ea011fb169c427932b8680f2ccf after we talked offline and we could use collapse. .
I read up some more on collapse and it seems that it is quite resource intensive and generally should be used sparsely and especially if the search request is not made frequently. The agent status API would be an exception here. Just thought I'll note it here in case we see performance issues cause of this in the future.

Reply:
Lets discuss if you still have questions/doubts.